pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -976,14 +1115,54 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 			return a.deleteNode(ctx, node)
 		}
 
-		// node is deleted, we can power on the host
+		//we need this annotation to differentiate between unhealthy machine that
+		//needs remediation, and an unhealthy machine that just got remediated
+		return a.addPoweredOffForRemediationAnnotation(ctx, machine)
+	}
+
+	// here we know that host has been powered off and node has been deleted
+	if hasPowerOffRequestAnnotation(baremetalhost) {
+		// we can now power host back on
 		log.Printf("Requesting Host %s power on for Machine %s",
 			baremetalhost.Name, machine.Name)
-		return a.requestPowerOn(ctx, baremetalhost)
+		return a.requestPowerOn(ctx, machine, baremetalhost)
 	}
 
+	node, err := a.getNodeByMachine(ctx, machine)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
+			return err
+		}
+	}
 	//node is still not running, so we requeue
 	if node == nil {
+		if !isMaster(machine) {
+			mhc = a.getMhcByMachine(machine)
+			timeout := powerOnDefaultTimeout
+			if mhc != nil {
+				if duration, err := time.ParseDuration(mhc.Spec.NodeStartupTimeout); err == nil {
```

Comment:
`mhc.Spec.NodeStartupTimeout` can be empty string, and this is a valid case.
In that case the log output might be a bit confusing.

Reply:
ack